### PR TITLE
Make fv_stations materialized

### PR DESCRIPTION
**Pull request by jdk24**
## Problem
Thanks to ever changing requirements, fv_stations now does a lot of heavy lifting, takes up to 4 seconds to respond.
This makes "some" clients time out before getting a response. 

## Solution 
Minimal changes: Make the view **materialized** in `015_create_wms_call_view.sql` (see caveats), refresh it with `pg_cron` like we already do for `station_data.fv_station_metadata` in `007_create_pg_cron_jobs.sql`.

## Caveats

Cannot do `DROP VIEW IF EXISTS fv_stations` in case it's already a MATERIALIZED VIEW -> Throws error -> DB launch fails. Added small helper function to drop whatever kind of VIEW fv_stations is. 

## Optional 
 
We could drop the new function straight afterwards.

### CODE REVIEW

File: db/update-files.d/015_create_wms_call_view.sql [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
 */
 
-DROP VIEW IF EXISTS station_data.fv_stations;
+CREATE OR REPLACE FUNCTION station_data.drop_view(target_schema TEXT, target_view text)
+ RETURNS void
+ LANGUAGE plpgsql
+AS $function$  
+BEGIN
+    IF EXISTS (SELECT matviewname from pg_matviews where schemaname = target_schema and matviewname = target_view) THEN
+        RAISE NOTICE 'Dropped materialized view %', target_view;
```
Comment (JakobMiksch) [inline db/update-files.d/015_create_wms_call_view.sql:16]: maybe switch the order of both statements -  I guess the message should be show after the drop has actually happened

File: db/update-files.d/015_create_wms_call_view.sql [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
+        EXECUTE format('DROP MATERIALIZED VIEW %I.%I', target_schema,target_view); 
+    ELSEIF EXISTS (SELECT viewname from pg_views where schemaname = target_schema and viewname = target_view) THEN
+        RAISE NOTICE 'Dropped view %', target_view;
```
Comment (JakobMiksch) [inline db/update-files.d/015_create_wms_call_view.sql:19]: maybe switch the order of both statements - I guess the message should be show after the drop has actually happened

File: db/update-files.d/007_create_pg_cron_jobs.sql [hunk 3]
```diff
@@ -11,6 +11,12 @@ constraint on table cron.job (with unkown results).
 To add queries, alter cols 1 and 2 (cron schedule and query).
 */
 
+-- Refresh steation metainfo for station viewer at 11:00 every day
```
[the remark this answers is not in the document]
Comment (jdk24) [reply db/update-files.d/007_create_pg_cron_jobs.sql]: Good catch. Changed all to name of view.

File: db/update-files.d/015_create_wms_call_view.sql (hunk 1 above)
Comment (jdk24) [reply db/update-files.d/015_create_wms_call_view.sql:17]: Good idea. Done.

File: db/update-files.d/007_create_pg_cron_jobs.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (jdk24) [reply db/update-files.d/007_create_pg_cron_jobs.sql]: Added 11:00am to clarify

**Review by chrismayer — APPROVED** [review]
Only went through the code and did not test running. LGTM.

**Comment by JakobMiksch** [comment]
@jdk24 thanks for the explanation. It is ready to merge from my side.

**Pull request merged**